rest/model/test.go

```diff
@@ -29,139 +28,72 @@ type APITest struct {
 	EndTime         *time.Time `json:"end_time"`
 	Duration        float64    `json:"duration"`
 	LogTestName     *string    `json:"log_test_name,omitempty"`
-	LineNum         int        `json:"line_num,omitempty"`
 }
 
 // TestLogs is a struct for storing the information about logs that will be
 // written out as part of an APITest.
 type TestLogs struct {
-	URL            *string `json:"url"`
-	LineNum        int     `json:"line_num"`
-	URLRaw         *string `json:"url_raw"`
-	LogId          *string `json:"log_id"`
-	RawDisplayURL  *string `json:"url_raw_display"`
-	HTMLDisplayURL *string `json:"url_html_display"`
+	URL     *string `json:"url"`
+	URLRaw  *string `json:"url_raw"`
+	LineNum int     `json:"line_num"`
+	LogId   *string `json:"log_id,omitempty"`
 }
 
 func (at *APITest) BuildFromService(st interface{}) error {
 	switch v := st.(type) {
 	case *testresult.TestResult:
+		at.Id = utility.ToStringPtr(v.ID.Hex())
 		at.Execution = v.Execution
-		at.LineNum = v.LineNum
+		if v.GroupID != "" {
+			at.GroupId = utility.ToStringPtr(v.GroupID)
+		}
 		at.Status = utility.ToStringPtr(v.Status)
 		at.TestFile = utility.ToStringPtr(v.TestFile)
 		if v.DisplayTestName != "" {
 			at.DisplayTestName = utility.ToStringPtr(v.DisplayTestName)
 		}
 		at.ExitCode = v.ExitCode
-		at.Id = utility.ToStringPtr(v.ID.Hex())
-
 		startTime := utility.FromPythonTime(v.StartTime)
 		endTime := utility.FromPythonTime(v.EndTime)
 		at.Duration = v.EndTime - v.StartTime
 		at.StartTime = ToTimePtr(startTime)
 		at.EndTime = ToTimePtr(endTime)
 
+		tr := task.ConvertToOld(v)
 		at.Logs = TestLogs{
-			URL:     utility.ToStringPtr(v.URL),
-			URLRaw:  utility.ToStringPtr(v.URLRaw),
-			LogId:   utility.ToStringPtr(v.LogID),
+			URL:     utility.ToStringPtr(tr.GetLogURL(false)),
+			URLRaw:  utility.ToStringPtr(tr.GetLogURL(true)),
 			LineNum: v.LineNum,
 		}
-
-		isEmptyLogID := v.LogID == ""
-		isEmptyURL := v.URL == ""
-		isEmptyURLRaw := v.URLRaw == ""
-
-		if !isEmptyURL {
-			at.Logs.HTMLDisplayURL = at.Logs.URL
-		} else if isEmptyLogID {
-			at.Logs.HTMLDisplayURL = utility.ToStringPtr(fmt.Sprintf(
-				"/test_log/%s/%d?test_name=%s&group_id=%s#L%d",
-				url.PathEscape(v.TaskID),
-				v.Execution,
-				url.QueryEscape(v.TestFile),
-				url.QueryEscape(v.GroupID),
-				v.LineNum,
-			))
-		} else {
-			dispString := fmt.Sprintf("/test_log/%s#L%d", *at.Logs.LogId, at.Logs.LineNum)
-			at.Logs.HTMLDisplayURL = &dispString
-		}
-
-		if !isEmptyURLRaw {
-			at.Logs.RawDisplayURL = at.Logs.URLRaw
-		} else if isEmptyLogID {
-			at.Logs.RawDisplayURL = utility.ToStringPtr(fmt.Sprintf(
-				"/test_log/%s/%d?test_name=%s&group_id=%s&text=true",
-				url.PathEscape(v.TaskID),
-				v.Execution,
-				url.QueryEscape(v.TestFile),
-				url.QueryEscape(v.GroupID),
-			))
-		} else {
-			dispString := fmt.Sprintf("/test_log/%s?text=true", *at.Logs.LogId)
-			at.Logs.RawDisplayURL = &dispString
+		if v.LogID != "" {
```

Comment:
What's the significance of not allowing LogID to be an empty string?

Reply:
I just don't want to clutter the API response. LogID is only really useful when fetching test logs that are stored in evg (which `GetLogURL` already takes care of since it populates the url fields).